src/handlers.cpp

```diff
@@ -2506,6 +2440,63 @@ void PushProfileTask::update_reg_data()
   SAS::report_event(event);
 }
 
+void PushProfileTask::get_registration_set()
```

Comment:
Can you use irs or implicit_registration_set consistently in all your variable/function names?